x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -683,4 +696,139 @@ private XContentBuilder getTokenIndexMappings() {
         }
     }
 
+    private SystemIndexDescriptor getSecurityProfileIndexDescriptor() {
+        return SystemIndexDescriptor.builder()
+            .setIndexPattern(".security-profile-[0-9]+*")
+            .setPrimaryIndex(INTERNAL_SECURITY_PROFILE_INDEX_8)
+            .setDescription("Contains user profile documents")
+            .setMappings(getProfileIndexMappings())
+            .setSettings(getProfileIndexSettings())
+            .setAliasName(SECURITY_PROFILE_ALIAS)
+            .setIndexFormat(INTERNAL_PROFILE_INDEX_FORMAT)
+            .setVersionMetaKey(SECURITY_VERSION_STRING)
+            .setOrigin(SECURITY_ORIGIN)
+            .setThreadPools(ExecutorNames.CRITICAL_SYSTEM_INDEX_THREAD_POOLS)
+            .build();
+    }
+
+    private Settings getProfileIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
+            .put(IndexMetadata.SETTING_PRIORITY, 1000)
+            .put("index.refresh_interval", "1s")
+            .put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), INTERNAL_PROFILE_INDEX_FORMAT)
+            .build();
+    }
+
+    private XContentBuilder getProfileIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field(SECURITY_VERSION_STRING, Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "strict");
+                builder.startObject("properties");
+                {
+                    builder.startObject("uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("redirect_uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
```

Comment:
This is a new field that is not in the tech spec. I think it could be useful for migration purpose, e.g. a separate profile was somehow created for the same person and already used in case assignment etc. A lightweight migration can be setting this field to point to the old profile document id. Happy to drop this field if it feels like pre-mature optimisation.

Reply:
I don't have enough information to tell how I feel about it. If this has been discussed before, please let me know.
Otherwise, we can discuss now, or at any other time you require my input.

I undestand that this is used to "link" two different profiles across different security domains. I would also like to understand when/how such profiles can be linked, un-linked, and how listing and searching works for linked profiles.

Reply:
I am happy to drop the field since it does not worth to stall the PR over it. We can always add it later if at all.